#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.15839793342993452 0.27282805009541417 0.9489335855402422 1.0981506595856745
  position 0.12395184280793275 -0.953027527124739 0.39165024061818443
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
}
Robot {
  translation 0.6333618333914517 -0.029361689880447305 -8.98037222497941e-05
  rotation -1.65746384132225e-05 -0.0002580333235956499 -0.9999999665720422 0.13688379079584317
  children [
    InertialUnit {
    }
    GPS {
    }
    Transform {
      translation 0.045 0 0.005
      children [
        DEF ball Shape {
          appearance PBRAppearance {
          }
          geometry Sphere {
            radius 0.005
            subdivision 2
          }
        }
      ]
    }
    Transform {
      translation -0.045 0 0.005
      children [
        DEF ball Shape {
          appearance PBRAppearance {
          }
          geometry Sphere {
            radius 0.005
            subdivision 2
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 11.837718959812817
        axis 0 1 0
        anchor 0 0.055 0.02
      }
      device [
        PositionSensor {
          name "sensor_izquierdo"
        }
        RotationalMotor {
          name "motor_izquierdo"
          maxVelocity 30
        }
      ]
      endPoint Solid {
        translation 0 0.055 0.02
        rotation 0.8801596035511481 -0.3356473183518112 0.33564855125551135 1.6981067137806463
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.8 0 0
              metalness 0.5
            }
            geometry DEF left Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE left
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 8.407766037195612
        axis 0 1 0
        anchor 0 -0.055 0.02
      }
      device [
        PositionSensor {
          name "sensor_derecho"
        }
        RotationalMotor {
          name "motor_derecho"
          maxVelocity 30
        }
      ]
      endPoint Solid {
        translation 0 -0.055 0.02
        rotation -0.3666870676180315 -0.6578515847569023 0.6578540011845494 3.844515833226855
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.8 0 0
              metalness 0.5
            }
            geometry DEF left Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE left
        physics Physics {
        }
      }
    }
    Transform {
      translation 0 0 0.025
      children [
        DEF body Shape {
          appearance PBRAppearance {
            baseColor 0.458824 0.313726 0.482353
            metalness 0.5
          }
          geometry Cylinder {
            height 0.04
            radius 0.05
          }
        }
      ]
    }
  ]
  name "robot(1)"
  boundingObject Group {
    children [
      Transform {
        translation 0.045 0 0.005
        children [
          USE ball
        ]
      }
      Transform {
        translation -0.045 0 0.005
        children [
          USE ball
        ]
      }
      Transform {
        translation 0 0 0.025
        children [
          USE body
        ]
      }
    ]
  }
  physics Physics {
    density -1
    mass 0.31415926535897937
    centerOfMass [
      0 0 0.025
    ]
    inertiaMatrix [
      0.00043458698374658825 0.00043458698374658825 0.00039269908169872427
      0 0 0
    ]
  }
  controller "controlador"
}
